### DISCUSSION TITLE
CPU & RAM Optimisation

### DISCUSSION THREAD

**Discussion by harshakhmk** [qdrant/qdrant#3514] [General]
Hi, I am looking for ways to optimise CPU RAM Optimisation of Qdrant server, as I have run the server on low end computing machines
Are there any ways to disable certain modules during build to optimise the memory consumption

**Comment by generall** [comment]
This might be relevant https://qdrant.tech/articles/memory-consumption/

**Reply by harshakhmk** [reply]
I have seen the article, but I am looking for options to optimise CPU and memory required during server idle time, as I have to run the application around 100MB
Can you comment on specific ways to disable certain modules during build or runtime if feasible
Additionally if there are any other ways other than mentioned in the article to run the server for low end machines

**Reply by harshakhmk** [reply]
Hi, any leads on this, would be highly appreciated & helpful
Thanks in advance

**Reply by harshakhmk** [reply]
Like updating config files or compiling only specific modules of code to run the application in lower end machines
I am interested to know what would be the ideal configurations in config file for an lower end arm architecture machine

**Reply by harshakhmk** [reply]
any build flags or runtime flags are there to achieve the optimisation part in terms of CPU & RAM and any other optimisation

**Reply by harshakhmk** [reply]
Kindly comment @generall

**Reply by generall** [reply]
i don't think there is a way to force memory deallocation during idle other than just limiting it. Also, the RssFile memory can usually be shared with other processes

**Reply by harshakhmk** [reply]
I am interested to know what would be the ideal configurations in config file for an lower end arm architecture machine

**Reply by generall** [reply]
how much data do you plan to serve there?

**Reply by harshakhmk** [reply]
I am trying to run the server under 100 MB, and I am experimenting with dataset of 60000 vectors of 785 dimensions

**Reply by harshakhmk** [reply]
Hi @generall , any comments to optimise resource utilisation for a low end computing machine with a goal to run under 100 MB with a dataset of 20000-60000 with 785 dimensions

**Reply by generall** [reply]
I run `docker run --rm -it --memory=100Mb --network=host  qdrant/qdrant:v1.7.4` and it can work pretty much as-is with 60k vectors. You only need to enable on_disk=True for them

**Comment by harshakhmk** [comment]
> Hi @generall , any comments to optimise resource utilisation for a low end computing machine with a goal to run under 100 MB with a dataset of 20000-60000 with 785 dimensions
Could you suggest ideal configuration settings for the above described usecase 

To enable on_disk= true, should I add it in client SDK code

**Reply by generall** [reply]
yes

**Reply by harshakhmk** [reply]
Hi @generall @dzhao @bazhenov @yarikoptic  I was asking for ideal configuration settings for a low end computing machine with a goal to run under 100 MB

**Reply by timvisee** [reply]
> Hi @generall @dzhao @bazhenov @yarikoptic  I was asking for ideal configuration settings for a low end computing machine with a goal to run under 100 MB

It'd be best to test this yourself.

But you likely want;
- everything on disk
- only one collection with one shard

Though, just using 100MB of memory will likely make it perform quite badly.
